cdap-unit-test/src/test/java/co/cask/cdap/test/app/ServiceLifeCycleTestRun.java

```diff
@@ -177,6 +181,8 @@ public Boolean call() throws Exception {
             .equals(ImmutableList.copyOf(newStates.get(lastStates.keySet().iterator().next())));
         }
       }, 10, TimeUnit.SECONDS, 100, TimeUnit.MILLISECONDS);
+      serviceManager.stop();
+      serviceManager.waitForRun(ProgramRunStatus.KILLED, 10, TimeUnit.SECONDS);
```

Comment:
Maybe you can add a method to wait for all programs to stop running and call it in`TestBase#afterTest()` so that a lot of such stop and wait in individual tests can be removed